Kernel/Tasks/Process.h

```diff
@@ -228,8 +237,9 @@ class Process final
     static RefPtr<Process> from_pid_ignoring_jails(ProcessID);
     static SessionID get_sid_from_pgid(ProcessGroupID pgid);
 
-    SpinlockProtected<NonnullOwnPtr<KString>, LockRank::None> const& name() const;
-    void set_name(NonnullOwnPtr<KString>);
+    SpinlockProtected<NonnullRefPtr<TaskName>, LockRank::None> const& name() const;
+    ErrorOr<void> set_name(NonnullOwnPtr<KString>);
```

Comment:
We might want to put a bold difference this and the other method... something like `set_new_name` could be OK?